worlds/stick_ranger/__init__.py

```diff
@@ -0,0 +1,267 @@
+            opt_max = getattr(self.options, max_attr)
+            opt_min.value = low
+            opt_max.value = high
+            getattr(self.options, req_attr).value = rng.randint(low, high)
+
+    def generate_early(self) -> None:
+        self._validate_options()
+        self.excluded_locations: Set[str] = self._compute_excluded_locations()
+        self._generate_randomness()
+
+    def create_regions(self) -> None:
+        menu_region: Region = Region("Menu", self.player, self.multiworld)
+        world_map_region: Region = Region("World Map", self.player, self.multiworld)
+        self.multiworld.regions += [menu_region, world_map_region]
+        menu_to_world_map_exit: Entrance = Entrance(
+            self.player, "World Map", menu_region
+        )
+        menu_region.exits.append(menu_to_world_map_exit)
+        menu_to_world_map_exit.connect(world_map_region)
+
+        def filter_locations(
+            table: Dict[int, LocationDict], region: str, filter_func=None
+        ) -> Dict[str, int]:
+            return {
+                loc["name"]: loc_id
+                for loc_id, loc in table.items()
+                if loc["region"] == region
+                and (filter_func(loc) if filter_func else True)
+            }
+
+        def make_unlock_rule(region_name: str) -> bool:
+            return lambda state: state.has(f"Unlock {region_name}", self.player)
+
+        for region_name in regions:
+            region: Region = Region(region_name, self.player, self.multiworld)
+
+            stage_locs: Dict[str, int] = filter_locations(stages_table, region_name)
+            region.add_locations(stage_locs, SRLocation)
+
+            if self.options.shuffle_books.value == 1:
+                book_locs: Dict[str, int] = filter_locations(books_table, region_name)
+                region.add_locations(book_locs, SRLocation)
+
+            if self.options.shuffle_enemies.value > 0:
+                enemy_filters: Dict[int, Any] = {
+                    ENEMIES_OPTION_NON_BOSS: lambda loc: "boss"
+                    not in loc["name"].lower(),
+                    ENEMIES_OPTION_BOSS: lambda loc: "boss" in loc["name"].lower(),
+                    ENEMIES_OPTION_ALL: None,
+                }
+
+                enemy_locations: Dict[str, int] = filter_locations(
+                    enemies_table,
+                    region_name,
+                    enemy_filters.get(self.options.shuffle_enemies.value),
+                )
+                region.add_locations(enemy_locations, SRLocation)
+
+            self.multiworld.regions.append(region)
+
+            world_map_exit: Entrance = Entrance(
+                self.player, region_name, world_map_region
+            )
+            if region_name != "Opening Street":
+                world_map_exit.access_rule = make_unlock_rule(region_name)
+            world_map_region.exits.append(world_map_exit)
+            world_map_exit.connect(region)
```

Comment:
```suggestion
            world_map_exit = world_map_region.connect(region)
            if region_name != "Opening Street":
                world_map_exit.access_rule = make_unlock_rule(region_name)
```
Pretty sure this does the same thing